examples/DivisorCurve.jl

```diff
@@ -0,0 +1,327 @@
+
+function ==(D::CurveDivisor, E::CurveDivisor)
+    return D.C == E.C && D.divisor == E.divisor
+end
+
+################################################################################
+################################################################################
+
+@doc Markdown.doc"""
+    iseffective(D::CurveDivisor)
+
+Return `true` if `D` is an effective divisor, `false` otherwise.
+"""
+function iseffective(D::CurveDivisor)
+    return all(v >= 0 for v in values(D.divisor))
+end
+
+################################################################################
+################################################################################
+
+@doc Markdown.doc"""
+    multiplicity(C::AffinePlaneCurve, phi::AbstractAlgebra.Generic.Frac, P::Point)
+
+Return the multiplicity of `phi` on the curve `C` at the point `P`.
+"""
+function multiplicity(C::AffinePlaneCurve, phi::AbstractAlgebra.Generic.Frac, P::Point)
+    f = divrem(phi.num, C.eq)
+    g = divrem(phi.den, C.eq)
+    !iszero(g[2]) || error("This is not a rational function on `C`.")
+    if iszero(f[2])
+        return Inf
+    elseif isconstant(f[2])
+        a = 0
+    else
+        a = intersection_multiplicity(C, AffinePlaneCurve(f[2]), P)
+    end
+    if isconstant(g[2])
+        b = 0
+    else
+        b = intersection_multiplicity(C, AffinePlaneCurve(g[2]), P)
+    end
+    return a-b
+end
+
+###############################################################################
+
+@doc Markdown.doc"""
+    multiplicity(C::AffinePlaneCurve{S}, F::Oscar.MPolyElem{S}, P::Point{S}) where S <: FieldElem
+
+Return the multiplicity of `F` on the curve `C` at the point `P`.
+"""
+function multiplicity(C::AffinePlaneCurve{S}, F::Oscar.MPolyElem{S}, P::Point{S}) where S <: FieldElem
+    f = divrem(F, C.eq)
+    if iszero(f[2])
+        return Inf
+    elseif isconstant(f[2])
+        return 0
+    else
+        return intersection_multiplicity(C, AffinePlaneCurve(f[2]), P)
+    end
+end
+
+################################################################################
+
+@doc Markdown.doc"""
+    multiplicity(C::ProjPlaneCurve{S}, F::Oscar.MPolyElem{S}, P::Oscar.Geometry.ProjSpcElem{S}) where S <: FieldElem
+
+Return the multiplicity of `F` on the curve `C` at the point `P`.
+"""
+function multiplicity(C::ProjPlaneCurve{S}, F::Oscar.MPolyElem{S}, P::Oscar.Geometry.ProjSpcElem{S}) where S <: FieldElem
+    f = divrem(F.f, defining_equation(C))
+    if iszero(f[2])
+        return Inf
+    elseif isconstant(f[2])
+        return 0
+    else
+        R = parent(C.eq)
+        return intersection_multiplicity(C, ProjPlaneCurve(R(f[2])), P)
+    end
+end
+
+################################################################################
+################################################################################
+
+@doc Markdown.doc"""
+    divisor(C::AffinePlaneCurve{S}, F::Oscar.MPolyElem{S}) where S <: FieldElem
+
+Return the divisor on the curve `C` defined by `F`.
```

Comment:
```suggestion
Return the divisor defined by the polynomial `F` on the curve `C`.
```
I'd reorder this, to (a) avoid ambiguity ("the curve defined by F"), and (b) to be consistent with preceding docstrings.